qiskit_dynamics/pulse_simulator/pulse_simulator.py

```diff
@@ -0,0 +1,526 @@
+      proper probability distributions.
+    * ``meas_level``: Form of measurement return. Only supported value is ``2``, indicating that
+      counts should be returned. Defaults to ``meas_level==2``.
+    * ``max_outcome_level``: For ``meas_level==2``, the maximum outcome for each subsystem.
+      Values will be rounded down to be no larger than ``max_outcome_level``. Must be a positive
+      integer or ``None``. If ``None``, no rounding occurs. Defaults to ``1``.
+    * ``memory``: Boolean indicating whether to return a list of explicit measurement outcomes for
+      every experimental shot. Defaults to ``True``.
+    * ``seed_simulator``: Seed to use in random sampling. Defaults to ``None``.
+    * ``experiment_result_function``: Function for computing the ``ExperimentResult``
+      for each simulated experiment. This option defaults to
+      :func:`default_experiment_result_function`, and any other function set to this option
+      must have the same signature. Note that the default utilizes various other options that
+      control results computation, and hence changing it will impact the meaning of other options.
+    """
+
+    def __init__(
+        self,
+        solver: Solver,
+        target: Optional[Target] = None,
+        provider: Optional[Provider] = None,
+        **options,
+    ):
+        """Instantiate with a :class:`.Solver` instance and additional options.
+
+        Args:
+            solver: Solver instance configured for pulse simulation.
+            target: Target object.
+            provider: Provider of the backend.
+            options: Additional configuration options for the simulator.
+        Raises:
+            QiskitError: If any instantiation arguments fail validation checks.
+        """
+
+        super().__init__(
+            name="PulseSimulator",
+            description="Pulse enabled simulator backend.",
+            backend_version=0.1,
+            provider=provider,
+        )
+
+        # Dressed states of solver, will be calculated when solver option is set
+        self._dressed_evals = None
+        self._dressed_states = None
+        self._dressed_states_adjoint = None
+
+        # add subsystem_dims to options so set_options validation works
+        if "subsystem_dims" not in options:
+            options["subsystem_dims"] = [solver.model.dim]
+
+        # Set simulator options
+        self.set_options(solver=solver, **options)
+
+        if self.options.subsystem_labels is None:
+            labels = list(range(len(self.options.subsystem_dims)))
+            self.set_options(subsystem_labels=labels)
+
+        if self.options.meas_map is None:
+            meas_map = [[idx] for idx in self.options.subsystem_labels]
+            self.set_options(meas_map=meas_map)
+
+        # self._target = target or Target() doesn't work as bool(target) can be False
+        if target is None:
+            target = Target()
+
+        # add default simulator measure instructions
```

Comment:
Not urgent, but if we decided to do more complicated things later with default target we could move this to an internal method for for initializing the target.

Reply:
Sounds good - for now I'll leave it as is.